crates/compiler/parse/src/header.rs

```diff
@@ -124,40 +124,61 @@ impl<'a> ExposedName<'a> {
     }
 }
 
+pub trait Keyword: Copy + Clone + Debug {
+    const KEYWORD: &'static str;
+}
+
+macro_rules! keywords {
+    ($($name:ident => $string:expr),* $(,)?) => {
+        $(
+            #[derive(Copy, Clone, PartialEq, Eq, Debug)]
+            pub struct $name;
+
+            impl Keyword for $name {
+                const KEYWORD: &'static str = $string;
+            }
+        )*
+    }
+}
```

Comment:
why is this a trait. You could add the associated constant to the struct directly

```
impl $name {
    const KEYWORD: &'static str = $string;
}
```

I could not find actual usages of the trait, but maybe being generic over keywords is helpful somewhere?

Reply:
I actually use that in `spaces_around_keyword` - it takes a generic type `K: Keyword`, and uses `K::KEYWORD` to decide what to match in the input.

I _wanted_ to use this in `fmt` to generate `Formattable` impls for all of these - but alas `impl<K: Keyword> Formattable for K` ends up violating the rules about having non-overlapping impls (it conflicts with `impl<'a, F: Formattable> Formattable for &'a F`.